elastic/eui
# React 18 upgrade

## Summary

This is the first PR upgrading to React 18 and bumping all dependency versions that need it in order to support React 18.

Please note that the unit and e2e tests are not 100% passing. This is expected and will be addressed in further PRs to the `feature/react-18` feature branch.

- [x] Upgrade React and ReactDOM to version 18.2.0 and other related packages
- [x] Switch to using the hacky `@cfaester/enzyme-adapter-react-18` enzyme adapter for React 18
- [x] Update component types to support React 18
- [x] Update jest emotion serializer to fix class name transformation
- [x] Migrate EuiPortal tests to RTL
- [x] Integrate Redux DevTools for easy debugging and remove react-router-redux as it doesn't seem to be used or maintained anymore
- [x] Upgrade Cypress renderer to @cypress/react18
- [x] Rewrite EuiPortal to be a function component and use hooks to manage the container element

## QA

1. Pull the changes and perform a clean yarn install (`rm -rf node_modules && yarn`)
2. Run `yarn start` and check if the development environments starts up and is accessible in browser
3. Open `http://localhost:8030` and verify the application is (more or less) working as intended - there are no hard crashes and components are behaving as they would normally do
4. Run `yarn build` and check if the production build is passing
5. Run `yarn lint` to confirm there are no typing or code style issues

Ignore failing unit and e2e tests for now. They will be fixed in the following PRs to `feature/react-18`.

### General checklist


Changelog and possible breaking changes will be listed when merging the feature branch (`feature/react-18`) to main
